src/ProxyClient/Fastly.php

```diff
@@ -0,0 +1,158 @@
+<?php
+
+/*
+ * This file is part of the FOSHttpCache package.
+ *
+ * (c) FriendsOfSymfony <http://friendsofsymfony.github.com/>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace FOS\HttpCache\ProxyClient;
+
+use FOS\HttpCache\ProxyClient\Invalidation\ClearCapable;
+use FOS\HttpCache\ProxyClient\Invalidation\PurgeCapable;
+use FOS\HttpCache\ProxyClient\Invalidation\RefreshCapable;
+use FOS\HttpCache\ProxyClient\Invalidation\TagCapable;
+use Symfony\Component\HttpFoundation\Request;
+
+/**
+ * Fastly HTTP cache invalidator.
+ *
+ * @author Simone Fumagalli <simone.fumagalli@musement.com>
+ */
+class Fastly extends HttpProxyClient implements ClearCapable, PurgeCapable, RefreshCapable, TagCapable
+{
+    /**
+     * @internal
+     */
+    const HTTP_METHOD_PURGE = 'PURGE';
+
+    /**
+     * {@inheritdoc}
+     *
+     * @see https://docs.fastly.com/api/purge#purge_db35b293f8a724717fcf25628d713583
+     */
+    public function invalidateTags(array $tags)
+    {
+        $headers = [
+            'Fastly-Key' => $this->options['authentication_token'],
+            'Accept' => 'application/json',
+        ];
+
+        if (true === $this->options['soft_purge']) {
+            $headers['Fastly-Soft-Purge'] = 1;
+        }
+
+        foreach (\array_chunk($tags, 256) as $tagChunk) {
```

Comment:
how do we know this number? i think in other clients, we did this more sophisticated with the actual header length. i'd assume we need to do the same here, to avoid over long headers when purging lots of tags with long names.

Reply:
It’s in their docs as hard limit, I'll add a link.

Reply:
but if each tag would be 40 characters (because i use bad tags, or uuid or something), we would end up with a 10k header line, no? that is likely to break, depending on intermediary systems.

Reply:
RTFM ;)
Added link now